File: Excel/459 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/459 Next Perfect Square.xlsx", range = "A1:A10")
test  = read_excel("Excel/459 Next Perfect Square.xlsx", range = "B1:B10")

find_next_perf_square = function(n) (floor(sqrt(n)) + 1) ** 2

result = input %>%
  mutate(`Answer Expected` = map_dbl(Number, find_next_perf_square)) %>%
  select(-Number)

identical(result, test)
# [1] TRUE

File: Excel/400-499/459/459 Challenge.py
import pandas as pd
import math

input = pd.read_excel("459 Next Perfect Square.xlsx", usecols="A")
test  = pd.read_excel("459 Next Perfect Square.xlsx", usecols="B")

def find_next_perf_square(n):
    return (math.floor(math.sqrt(n)) + 1) ** 2

result = input["Number"].apply(find_next_perf_square)

print(result.equals(test["Answer Expected"])) # True